### ISSUE TITLE
How to use --quality with both audio and video ITAGs

### ISSUE THREAD

**Issue by PacoH** [fent/node-ytdl#77]
I'm trying to specify both audio and video format quality. What is the syntax for more than one ITAG?

I tried `-q 140,298` and it is the only way that was accepted but it only downloaded the audio track.

Both tracks exist according to the `-i` option.

**Comment by fent** [comment]
The `--quality` option downloads only one format, the first it can find.

If you're trying to merge two formats, that's something [pully](https://github.com/JimmyBoh/pully) can do

**Comment by PacoH** [comment]
Yes that's exactly what I want to do. I want higher quality than the default, especially since `youtube-dl` is broken now––downloads are slowed to a few 100kB/s.

`pully` seemed to be exactly what I was looking for until I installed and tried to run it. All I get is:

	$ pully
	env: node\r: No such file or directory

Any ideas?

**Comment by fent** [comment]
I don't know :/

I see you already opened an issue on pully

**Comment by PacoH** [comment]
OK, after hours of reading various posts about the problem, I found the solution––[osx cli issue · Issue #9 · JimmyBoh/pully](https://github.com/JimmyBoh/pully/issues/9#issuecomment-298659982)––buried deep within a thread. The problem, which was supposed to have been fixed at the time of the thread, 6 months ago, was/is freakin' `DOS` carriage returns, `\r`, in the code, which were not being recognized by OS X's BSD Unix. The fix is simple but was hard to find. `dos2unix` removed the `\r`s and now it works just fine. I'm not at all happy with its speed. It took just as long as `youtube-dl` did for a 21.7MB file, 4+ min, for a smaller, lower quality 16.9MB file. `ytdl` took 17.6 sec for a bigger, 19MB file. So `ytdl` is still the speed king by an incredible margin. If only it could select BOTH streams and mux them using `ffmpeg`, it would be the undisputed champion, and a replacement for `youtube-dl` if it doesn't get fixed.
